pkg/remotewrite/remotewrite_test.go

```diff
@@ -220,3 +221,55 @@ func TestNewSeriesWithNativeHistogramMeasure(t *testing.T) {
 	require.True(t, ok)
 	assert.NotNil(t, nhs.H)
 }
+
+func TestOutputSetTrendStatsResolver(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		stats           []string
+		expResolverKeys []string
+	}{
+		{
+			stats:           []string{},
+			expResolverKeys: []string{},
+		},
+		{
+			stats:           []string{"sum"},
+			expResolverKeys: []string{"sum"},
+		},
+		{
+			stats:           []string{"avg"},
+			expResolverKeys: []string{"avg"},
+		},
+		{
+			stats:           []string{"p(27)", "p(0.999)", "p(1)", "p(0)"},
+			expResolverKeys: []string{"p27", "p0999", "p1", "p0"},
+		},
+		{
+			stats: []string{
+				"count", "sum",
+				"max", "min", "med", "avg", "p(90)", "p(99)",
+			},
+			expResolverKeys: []string{
+				"count", "sum",
+				"max", "min", "med", "avg", "p90", "p99",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		o := Output{}
+		err := o.setTrendStatsResolver(tt.stats)
+		require.NoError(t, err)
+		require.NotNil(t, o.trendStatsResolver)
+
+		assert.Len(t, o.trendStatsResolver, len(tt.expResolverKeys))
+		assert.ElementsMatch(t, tt.expResolverKeys, func() []string {
+			var keys []string
+			for statKey := range o.trendStatsResolver {
```

Comment:
I think you could use `copy` here instead. That would make the code even shorter?